tutorials/workshop/Workshop_3_Generative_Model_Training.ipynb

```diff
@@ -0,0 +1,292 @@
+    "    n_epochs     A single pass through all the data in your training set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Computing O...\n",
+      "Estimating \\mu...\n",
+      "[500 epo]: TRAIN:[loss=0.001]\n",
+      "[1000 epo]: TRAIN:[loss=0.001]\n",
+      "[1500 epo]: TRAIN:[loss=0.001]\n",
+      "[2000 epo]: TRAIN:[loss=0.001]\n",
+      "[2500 epo]: TRAIN:[loss=0.001]\n",
+      "[3000 epo]: TRAIN:[loss=0.001]\n",
+      "[3500 epo]: TRAIN:[loss=0.001]\n",
+      "[4000 epo]: TRAIN:[loss=0.001]\n",
+      "[4500 epo]: TRAIN:[loss=0.001]\n",
+      "[5000 epo]: TRAIN:[loss=0.001]\n",
+      "Finished Training\n"
+     ]
+    }
+   ],
+   "source": [
+    "from snorkel.labeling.model.baselines import MajorityLabelVoter\n",
+    "from snorkel.labeling.model.label_model import LabelModel\n",
+    "from snorkel.model.utils import plusminus_to_categorical\n",
+    "\n",
+    "label_model = LabelModel(k=2, verbose=True)\n",
+    "label_model.train_model(\n",
+    "            plusminus_to_categorical(train_L.toarray()),\n",
+    "            lr = 1e-1,\n",
+    "            class_balance=[0.2,0.8],\n",
+    "            n_epochs=5000,\n",
+    "            log_train_every=500,\n",
+    "            seed=12\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Label Model Metrics\n",
+    "These are the weights learned for each LF"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Accuracy: 0.905\n",
+      "F1: 0.431\n",
+      "Precision: 0.373\n",
+      "Recall: 0.511\n",
+      "        y=1    y=2   \n",
+      " l=1    97     163   \n",
+      " l=2    93    2350   \n"
+     ]
+    }
+   ],
+   "source": [
+    "score = label_model.score((plusminus_to_categorical(dev_L.toarray()), \n",
+    "                           plusminus_to_categorical(dev_labels)), \n",
+    "                          metric=['accuracy','f1', 'precision', 'recall'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Majority Vote Comparison\n",
+    "We can also compare the performance of the LabelModel to computing a majority vote across all the LFs. In case of ties, we break ties randomly. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Accuracy: 0.450\n",
+      "F1: 0.164\n",
+      "Precision: 0.092\n",
+      "Recall: 0.768\n",
+      "        y=1    y=2   \n",
+      " l=1    146   1443   \n",
+      " l=2    44    1070   \n"
+     ]
+    }
+   ],
```

Comment:
TODO: Check why MV is doing this badly...

Reply:
added `break_ties = 2` and added explanation + random tie breaking comparison